openapi.yml

```diff
@@ -0,0 +1,578 @@
+openapi: 3.0.0
+info:
+  description: Preservation Catalog HTTP API
+  version: 1.0.0
+  title: Preservation Catalog HTTP API
+  license:
+    name: Apache 2.0
+    url: 'http://www.apache.org/licenses/LICENSE-2.0.html'
+servers:
+  - url: 'https://preservation-catalog-{env}-01.stanford.edu'
+    description: Production service
+    variables:
+      env:
+        default: prod
+  - url: 'https://preservation-catalog-{env}-01.stanford.edu'
+    description: Staging service
+    variables:
+      env:
+        default: stage
+  - url: 'https://preservation-catalog-{env}-01.stanford.edu'
+    description: QA service
+    variables:
+      env:
+        default: qa
+tags:
+  - name: catalog
+    description: Add and update MOABs in the catalog
+  - name: objects
+    description: Preserved Objects
+paths:
+  /v1/catalog:
+    post:
+      tags:
+        - catalog
+      summary: Add a new MOAB to the catalog
+      responses:
+        '201':
+          description: Created new MOAB
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/AuditResultsResponse'
+        '409':
+          description: Conflict - MOAB already exists
+        '406':
+          description: Invalid arguments passed
+        '500':
+          description: Internal server error
+      requestBody:
+        required: true
+        content:
+          application/x-www-form-urlencoded:
+            schema:
+              type: object
+              properties:
+                druid:
+                  $ref: '#/components/schemas/Druid'
+                incoming_version:
+                  type: integer
+                incoming_size:
+                  type: integer
+                storage_location:
+                  type: string
+                checksums_validated:
+                  type: string
+                  enum:
+                    - 'true'
+                    - 'True'
+                    - 'TRUE'
+                    - 'nil'
+                    - '1'
+                    - 'on'
+                    - 'false'
+                    - 'False'
+                    - 'FALSE'
+              required:
+                - druid
+                - incoming_version
+                - incoming_size
+                - storage_location
+  /v1/catalog/{druid}:
+    put:
```

Comment:
based on catalog_controller.rb#L28-29, I'm thinking we don't allow PUT?  Or perhaps we don't *mean* to allow it?

Reply:
I don't think that's the case. Will update that comment.